### PR TITLE
MethodSignatureSniffs: Add included and exluded patters

### PR DESCRIPTION
**Pull request by RiKap**
Hey.
I added `includedMethodPatterns` and `excludedMethodPatterns` property for `RequireSingleLineMethodSignatureSniff` and `RequireMultiLineMethodSignatureSniff`

**includedMethodPatterns**:
It will allow you to set methods which you would like to check by these two sniffs

**excludedMethodPatterns**:
It will allow you to set methods which you would like to ignore by these two sniffs

--------

Sometimes it is good to enforce `RequireMultiLineMethodSignatureSniff` just for one method, for example `__construct`.

Another benefit of this change is that you are able to use these two sniffs together.
It will force `RequireMultiLineMethodSignature` for `__construct` and `RequireSignleLineMethodSignature` for all other methods.
```
<rule ref="SlevomatCodingStandard.Classes.RequireMultiLineMethodSignature">
	<properties>
		<property name="minLineLength" type="int" value="0"/>
		<property name="includedMethodPatterns" type="array" value="
			/__construct(.*)/,
		"/>
	</properties>
</rule>
<rule ref="SlevomatCodingStandard.Classes.RequireSignleLineMethodSignature">
	<properties>
		<property name="minLineLength" type="int" value="0"/>
		<property name="excludedMethodPatterns" type="array" value="
			/__construct(.*)/,
		"/>
	</properties>
</rule>
```

There is no BC break. If you don't set `excludedMethodPatterns` and `includedMethodPatterns` it will works as before.

### CODE REVIEW

File: SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php [hunk 1]
```diff
@@ -60,4 +73,26 @@ public function process(File $phpcsFile, $methodPointer): void
...
 	}
 
+	/**
+	 * @param string $methodSignature
+	 * @param string[] $patterns
+	 * @return bool
+	 */
+	private function isMethodSignatureInPatterns(string $methodSignature, array $patterns): bool
+	{
+		foreach (SniffSettingsHelper::normalizeArray($patterns) as $pattern) {
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php]: Please use `normalizedPatters` so the method `SniffSettingsHelper::normalizeArray()` is not called everytime.

See Eg. https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Sniffs/Commenting/UselessFunctionDocCommentSniff.php#L130

File: SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php [hunk 2]
```diff
@@ -38,6 +44,13 @@ public function process(File $phpcsFile, $methodPointer): void
 		$signature = $this->getSignature($phpcsFile, $signatureStartPointer, $signatureEndPointer);
 		$signatureWithoutTabIndetation = $this->getSignatureWithoutTabs($signature);
 
+		if (count($this->includedMethodPatterns) !== 0) {
+			$isMethodSignatureInPatterns = $this->isMethodSignatureInPatterns($signatureWithoutTabIndetation, $this->includedMethodPatterns);
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php]: Use `FunctionHelper::getName()` and check only method names with patterns.

And please remove useless variable `$isMethodSignatureInPatterns` and join both `if` together.

File: README.md [hunk 3]
```diff
@@ -392,6 +392,8 @@ Sniff provides the following settings:
 
 * `includedMethodPatterns`: allows to configure which methods are included in sniff detection. This is an array of regular expressions (PCRE) with delimiters.
 
+* `excludedMethodPatterns`: allows to configure which methods are excluded from sniff detection. This is an array of regular expressions (PCRE) with delimiters.
+
```
Comment (kukulich) [inline README.md]: Please document how it works when you set both options together.

File: SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php [hunk 4]
```diff
@@ -60,4 +73,26 @@ public function process(File $phpcsFile, $methodPointer): void
...
+	 * @return bool
+	 */
+	private function isMethodSignatureInPatterns(string $methodSignature, array $patterns): bool
+	{
+		foreach (SniffSettingsHelper::normalizeArray($patterns) as $pattern) {
+			if (!SniffSettingsHelper::isValidRegularExpression($pattern)) {
+				throw new Exception(sprintf('%s is not valid PCRE pattern.', $pattern));
+			}
+
+			if (preg_match($pattern, $methodSignature) === 0) {
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php]: Use negative condition and `return true` please.

File: SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php [hunk 5]
```diff
@@ -60,4 +73,26 @@ public function process(File $phpcsFile, $methodPointer): void
...
+	/**
+	 * @param string $methodSignature
+	 * @param string[] $patterns
+	 * @return bool
+	 */
+	private function isMethodSignatureInPatterns(string $methodSignature, array $patterns): bool
+	{
+		foreach (SniffSettingsHelper::normalizeArray($patterns) as $pattern) {
+			if (!SniffSettingsHelper::isValidRegularExpression($pattern)) {
+				throw new Exception(sprintf('%s is not valid PCRE pattern.', $pattern));
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Classes/RequireSingleLineMethodSignatureSniff.php]: Please add unit test to test this line.

See `ForbiddenCommentsSniffTest`

**Comment by RiKap** [comment]
@kukulich 
2. Why it does not make sense now, but it did before?

**Comment by kukulich** [comment]
Now you check only method names.

**Comment by kukulich** [comment]
Thanks.

**Pull request merged**